indexing-hadoop/src/main/java/io/druid/indexer/Utils.java

```diff
@@ -123,4 +128,25 @@ public static void storeStats(
         stats
     );
   }
+
+  public static String getFailureMessage(Job failedJob, ObjectMapper jsonMapper)
+  {
+    try {
+      Map<String, String> taskDiagsMap = Maps.newHashMap();
+      TaskCompletionEvent[] completionEvents = failedJob.getTaskCompletionEvents(0, 100);
+      for (TaskCompletionEvent tce : completionEvents) {
+        String[] taskDiags = failedJob.getTaskDiagnostics(tce.getTaskAttemptId());
+        String combinedTaskDiags = "";
+        for (String taskDiag : taskDiags) {
+          combinedTaskDiags += taskDiag;
+        }
+        taskDiagsMap.put(tce.getTaskAttemptId().toString(), combinedTaskDiags);
+      }
+      return jsonMapper.writeValueAsString(taskDiagsMap);
+    }
+    catch (IOException | InterruptedException ie) {
+      log.error("couldn't get failure cause for job.");
```

Comment:
Please add the exception and job name to the log message.

Reply:
Added exception and job name